dotCMS/src/main/java/com/dotmarketing/common/business/journal/DistributedJournalAPIImpl.java

```diff
@@ -22,109 +21,130 @@
 	 */
+	@WrapInTransaction
 	public List<String> findCacheEntriesToRemove() throws DotDataException {
-		return distFac.findCacheEntriesToRemove();
+		return distributedJournalFactory.findCacheEntriesToRemove();
 	}
-	
+
+	@WrapInTransaction
 	public void addStructureReindexEntries(T structureInode) throws DotDataException {
-		distFac.addStructureReindexEntries(structureInode);
+		distributedJournalFactory.addStructureReindexEntries(structureInode);
 	}
-	
+
+	@WrapInTransaction
 	public synchronized void addBuildNewIndexEntries() throws DotDataException {
-		distFac.addBuildNewIndexEntries();
+		distributedJournalFactory.addBuildNewIndexEntries();
 	}
-	
+
+	@CloseDBIfOpened
 	public List<IndexJournal<T>> findContentReindexEntriesToReindex() throws DotDataException {
-		return distFac.findContentReindexEntriesToReindex();
+		return distributedJournalFactory.findContentReindexEntriesToReindex();
 	}
-	
+
+	@CloseDBIfOpened
 	public List<IndexJournal<T>> findContentReindexEntriesToReindex(boolean includeFailedRecords) throws DotDataException {
-		return distFac.findContentReindexEntriesToReindex(includeFailedRecords);
+		return distributedJournalFactory.findContentReindexEntriesToReindex(includeFailedRecords);
 	}
-	
+
+	@CloseDBIfOpened
 	public void processJournalEntries() throws DotDataException {
-		distFac.processJournalEntries();
+		distributedJournalFactory.processJournalEntries();
 	}
-	
+
+	@WrapInTransaction
 	public void deleteReindexEntryForServer(IndexJournal<T> iJournal) throws DotDataException {
-		distFac.deleteReindexEntryForServer(iJournal);
+		distributedJournalFactory.deleteReindexEntryForServer(iJournal);
 	}
 
-	public boolean isIndexationEnabled() {
-		return isIndexationEnabled();
+	@CloseDBIfOpened
+	public boolean areRecordsLeftToIndex() throws DotDataException {
+		return distributedJournalFactory.areRecordsLeftToIndex();
 	}
 
-	public void setIndexationEnabled(boolean indexationEnabled) {
-		setIndexationEnabled(indexationEnabled);
-	}
-	
-	public boolean areRecordsLeftToIndex() throws DotDataException {
-		return distFac.areRecordsLeftToIndex();
+	@Override
+	public boolean isIndexationEnabled() {
+		return false; // todo: should it exists
 	}
 
+	@CloseDBIfOpened
 	public long recordsLeftToIndexForServer() throws DotDataException {
 	    return recordsLeftToIndexForServer(DbConnectionFactory.getConnection());
 	}
 	
 	public long recordsLeftToIndexForServer(Connection conn) throws DotDataException {
-		return distFac.recordsLeftToIndexForServer(conn);
+		return distributedJournalFactory.recordsLeftToIndexForServer(conn);
 	}
-	
+
+	@WrapInTransaction
 	public void deleteLikeJournalRecords(IndexJournal<T> ijournal) throws DotDataException {
-		distFac.deleteLikeJournalRecords(ijournal);
+		distributedJournalFactory.deleteLikeJournalRecords(ijournal);
 	}
-	
+
+	@Override
+	public void setIndexationEnabled(boolean indexationEnabled) {
+		// todo: should it exists
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)